packages/@aws-cdk/aws-glue-alpha/lib/job.ts

```diff
@@ -743,6 +756,10 @@ export class Job extends JobBase {
       throw new Error('Both workerType and workerCount must be set');
     }
 
+    if (props.jobRunQueuingEnabled === true && props.maxRetries !== undefined && props.maxRetries > 0) {
```

Comment:
Please add condition `!Token.isUnresolved(props.maxRetries)`  to the if statement. 

This will enable support when `maxRetries` is token.

```suggestion
    if (props.jobRunQueuingEnabled === true && props.maxRetries !== undefined && Token.isUnresolved(props.maxRetries) && props.maxRetries > 0) {
```